sections/examples/dockerfile/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM node:14.8.0-alpine AS build
+
+# Copy dependency information and install all dependencies
+COPY --chown=node:node package.json package-lock.json ./
+
+RUN npm ci
+
+# Copy source code (and all other relevant files)
+COPY --chown=node:node src ./src
+
+# Build code
+RUN npm run build
+
```

Comment:
@goldbergyoni  - I love your work. 

I would add here `RUN npm run test`, goes with `"test": "node test.js"` in `package.json` and a `test/index.js` file that launches the server and hit it on `'/'`. 
You can also do it with `mocha` or `jest`, and use a `beforeAll` hook to launch the server and `afterAll` hook to terminate it. Checkout [e2e-helper](https://github.com/osher/e2e-helper)

A lot of value for not that much complexity :)

Reply:
@osher Thanks, happy to hear that words!

Reply:
@osher we had the debate on a previous issue about that. 

It is possible to run the tests in the Dockerfile but it's not what it is meant for. It is better practice to build an image with only the first stage (using the flag `--target build`) and then run testing, linting and other "checks" on this image. 
This allows to parallelize runs (each tool can run in its own container) and is more inline with the vision that Dockerfiles are for building images and docker containers are for running commands.
Depending on which tool you are using for your builds, it's also easier to see which step of the build failed since you have the step name instead of having to look at the Dockerfile build output to understand.

However we'll keep your example in mind when we'll work on a complete app example 👍